pkg/acs/calacs/lib/mkspt.c

```diff
@@ -167,53 +167,48 @@ int mkNewSpt (char *in_list, char *mtype, char *output) {
                 trlmessage (MsgText);
 		    }
 
-	        /* Update the FILENAME header keyword */
-            if (PutKeyStr (&header, "FILENAME", out_spt, "")) {
-                reportErrorAndFreeHeader (&header);
-                return (status = hstio_err());
+            /* Update the FILENAME header keyword */
+            if ((status = PutKeyStr (&header, "FILENAME", out_spt, ""))) {
+                return (status);
              }
 
-	        /* Update the ASN_MTYP header keyword */
-            if (PutKeyStr(&header, "ASN_MTYP", mtype, "Role of the Exposure in the Association")) {
-                reportErrorAndFreeHeader (&header);
-                return (status = hstio_err());
+            /* Update the ASN_MTYP header keyword */
+            if ((status = PutKeyStr(&header, "ASN_MTYP", mtype, "Role of the Exposure in the Association"))) {
+                return (status);
             }
 
-	        /* NOW, update the ROOTNAME header keyword */
-	        for (n = 0; n < strlen(rootname)-1; n++)
-	             rootname[n] = toupper(rootname[n]);
-            if (PutKeyStr (&header, "ROOTNAME", rootname, "")) {
-                reportErrorAndFreeHeader (&header);
-                return (status = hstio_err());
+            /* NOW, update the ROOTNAME header keyword */
+            for (n = 0; n < strlen(rootname)-1; n++)
+                rootname[n] = toupper(rootname[n]);
+            if ((status = PutKeyStr (&header, "ROOTNAME", rootname, ""))) {
+                return (status);
             }
 
             /* Update the OBSERVTN header keyword */
             strncpy (obsnum, &rootname[6], 3); obsnum[3] = '\0';
-            if (putKeyS (&header, "OBSERVTN", obsnum, "")) {
-                reportErrorAndFreeHeader (&header);
-                return (status = hstio_err());
+            if ((status = putKeyS (&header, "OBSERVTN", obsnum, ""))) {
+                return (status);
             }
 
-	        /* Update the NEXTEND header keyword to reflect number of input
-                images.  This will be updated as necessary at the end of
-				this routine.
+            /* Update the NEXTEND header keyword to reflect number of input
+               images.  This will be updated as necessary at the end of this routine.
             */
-            if (PutKeyInt (&header, "NEXTEND", nimgs, "")) {
-                reportErrorAndFreeHeader (&header);
-                return (status = hstio_err());
+            if ((status = PutKeyInt (&header, "NEXTEND", nimgs, ""))) {
+                return (status);
             }
 
             /* Write the new _spt.fits file */
             /* Open the image; this also writes the header */
             im = openOutputImage (out_spt, "", 0, &header, 0, 0, FITSBYTE);
-            if (hstio_err()) {
+            if (!im || hstio_err()) {
+                status = OPEN_FAILED;
```

Comment:
Just an FYI, since we're not going to call `WhichError()`, `status` can be set within in the return as it was, i.e. `return (status= OPEN_FAILED);`.